app/uk/gov/hmrc/fileupload/controllers/EnvelopeChecker.scala

```diff
@@ -20,46 +20,90 @@ import cats.data.Xor
 import play.api.Logger
 import play.api.http.Status._
 import play.api.libs.iteratee.Done
-import play.api.libs.json.Json
+import play.api.libs.json.{JsValue, Json}
 import play.api.libs.streams.Accumulator
 import play.api.mvc.Results._
 import play.api.mvc.{EssentialAction, RequestHeader, Result}
 import uk.gov.hmrc.fileupload.EnvelopeId
-import uk.gov.hmrc.fileupload.transfer.TransferService.{EnvelopeStatusNotFoundError, _}
+import uk.gov.hmrc.fileupload.quarantine.Constraints
+import uk.gov.hmrc.fileupload.transfer.TransferService._
 import uk.gov.hmrc.fileupload.utils.StreamsConverter
 
 import scala.concurrent.{ExecutionContext, Future}
 
 object EnvelopeChecker {
 
-  type WithValidEnvelope = EnvelopeId => EssentialAction => EssentialAction
+  type FileSize = Int
+  type WithValidEnvelope = EnvelopeId => (FileSize => EssentialAction) => EssentialAction
 
   import uk.gov.hmrc.fileupload.utils.StreamImplicits.materializer
 
-  def withValidEnvelope(check: (EnvelopeId) => Future[EnvelopeStatusResult])
+  val defaultFileSize = 10 * 1024 * 1024
+
+  def withValidEnvelope(checkEnvelopeDetails: (EnvelopeId) => Future[EnvelopeDetailResult])
                        (envelopeId: EnvelopeId)
-                       (action: EssentialAction)
+                       (action: FileSize => EssentialAction)
                        (implicit ec: ExecutionContext) =
     EssentialAction { implicit rh =>
       Accumulator.flatten {
-        check(envelopeId).map {
-          case Xor.Right("OPEN") =>
-            action(rh)
-          case Xor.Right(otherStatus) =>
-            logAndReturn(LOCKED, s"Unable to upload to envelope: $envelopeId with status: $otherStatus")
-          case Xor.Left(EnvelopeStatusNotFoundError(_)) =>
+        checkEnvelopeDetails(envelopeId).map {
+          case Xor.Right(envelope) =>
+            val status = (envelope \ "status").as[String]
+            if (status == "OPEN") {
```

Comment:
Probably better to convert to pattern matching instead of if-else statements on envelope.status